features/lwipstack/lwipopts.h

```diff
@@ -139,6 +139,26 @@
 
 #define LWIP_RAM_HEAP_POINTER       lwip_ram_heap
 
+// Number of simultaneously queued TCP segments.
+#ifdef MBED_CONF_LWIP_MEMP_NUM_TCP_SEG
```

Comment:
Why `#ifdef`

If we use Mbed compilation, these values always come from `mbed_lib.json`

Reply:
There was no setting process in `mbed_lib.json`. Thus, in order to set its customized value when coming from the `mbed_lib.json` and the default value when not coming, as with other elements, I added `#ifdef` process to this file.

Reply:
This is consistent with other settings - basically not setting it in the JSON lets lwIP do its own default, which sometimes is not a simple constant we could express in the JSON.

Or could we - I guess you could define something as `"5 * TCP_MSS"`. Maybe we could put all lwIP's defaults in as explicit JSON? I think it was only really those derived values that put us off doing that.

Reply:
@kjbracey-arm
Thank you for your comments, and may I ask you a bit?
What does `"5 * TCP_MSS"` indicate? is it related with the value of "tcp-snd-buf" and "tcp-wnd" ?
If so, I defined them as `"8 * TCP_MSS"`(is customize value for RZ/A1) in [json file](https://github.com/TomoYamanaka/mbed-os/blob/Improve_LWIP/features/lwipstack/mbed_lib.json#L153-L154).